src/mono/mono/mini/interp/transform.c

```diff
@@ -3180,6 +3181,7 @@ interp_inline_newobj (TransformData *td, MonoMethod *target_method, MonoMethodSi
 	push_var (td, dreg);
 	return TRUE;
 fail:
+	mono_metadata_free_mh (mheader);
```

Comment:
Shouldn't this be if (mheader) ...

Reply:
could be, there is an explicit null check in `mono_metadata_free_mh` though.